backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/question/domain/MultipleChoiceQuestion.java

```diff
@@ -0,0 +1,103 @@
+package pt.ulisboa.tecnico.socialsoftware.tutor.question.domain;
+
+import pt.ulisboa.tecnico.socialsoftware.tutor.answer.domain.QuestionAnswer;
+import pt.ulisboa.tecnico.socialsoftware.tutor.config.DateHandler;
+import pt.ulisboa.tecnico.socialsoftware.tutor.course.Course;
+import pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.TutorException;
+import pt.ulisboa.tecnico.socialsoftware.tutor.impexp.domain.Visitor;
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.dto.OptionDto;
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.dto.QuestionDto;
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import static pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.ErrorMessage.ONE_CORRECT_OPTION_NEEDED;
+import static pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.ErrorMessage.OPTION_NOT_FOUND;
+
+@Entity
+@DiscriminatorValue(Question.QuestionTypes.MultipleChoice)
+public class MultipleChoiceQuestion extends Question {
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "question",fetch = FetchType.EAGER, orphanRemoval=true)
```

Comment:
Why is it necessary orphanRemoval=true?

Reply:
Actually, I did not look much into it, I brought it over from the Question class. I will look it up to see what I can find.

Reply:
This is used so that when an Option is disconnected from the MultipleChoiceQuestion then that Option is also removed.